### PR TITLE
Create prometheus metric for item counts

### PR DESCRIPTION
**Pull request by holahmeds**
**Changes**
Create prometheus metric for number of items in library, by item type:
* Movie
* Series
* Season
* Episode
* MusicArtist
* MusicAlbum
* MusicVideo
* Audio
* Book
* PhotoAlbum
* Photo

**Issues**

 #3016

### CODE REVIEW

**Comment by holahmeds** [comment]
Hi all, first time contributor here, so if there is a better place to create this metric please let me know.

I plan to add more metrics (through subsequent PRs) as I get more familiar with the code.

**Comment by cvium** [comment]
It feels like something that should be in a plugin. I definitely don't like having Prometheus directly in LibraryManager.

**Comment by holahmeds** [comment]
The discussion in #2985 seems to indicate that all the planned metrics may not be possible with a plugin. It is more that a year old though, so if it is no longer true I can look into creating a plugin.

File: Emby.Server.Implementations/Metrics/PrometheusMetricsCollector.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PrometheusMetricsCollector" /> class.
+        /// </summary>
+        /// <param name="libraryManager">The library manager</param>
+        public PrometheusMetricsCollector(ILibraryManager libraryManager)
+        {
+            _libraryManager = libraryManager;
+            _libraryManager.ItemAdded += IncrementItemCount;
+            _libraryManager.ItemRemoved += (s, e) => UpdateItemCount();
```
Comment (holahmeds) [inline Emby.Server.Implementations/Metrics/PrometheusMetricsCollector.cs:42]: `ItemRemoved` only seems to provide the parent item for e.g. If a series is removed, it fires once for the series but not the episodes. Calling `GetRecursiveChildren()` on it does not give them either. Is there another way to get all individual items removed?

**Comment by holahmeds** [comment]
I have moved the metric out of Library manager and into a new class. All prometheus metrics can be put here.

**Review by robertas90poe — APPROVED** [review]

**Pull request closed without merging**